checker/src/org/checkerframework/checker/upperbound/UpperBoundVisitor.java

```diff
@@ -0,0 +1,131 @@
+package org.checkerframework.checker.upperbound;
+
+import com.sun.source.tree.ArrayAccessTree;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import java.util.ArrayList;
+import java.util.List;
+import javax.lang.model.element.ExecutableElement;
+import org.checkerframework.checker.upperbound.qual.LTLengthOf;
+import org.checkerframework.checker.upperbound.qual.LTOMLengthOf;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.common.basetype.BaseTypeVisitor;
+import org.checkerframework.dataflow.analysis.FlowExpressions;
+import org.checkerframework.framework.source.Result;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.javacutil.TreeUtils;
+
+/** Warns about array accesses that could be too high. */
+public class UpperBoundVisitor extends BaseTypeVisitor<UpperBoundAnnotatedTypeFactory> {
+
+    protected final List<ExecutableElement> IndexFirstArgListMethods;
```

Comment:
It seems like these methods should just be annotated like `add(@IndexFor("this") int i)`.  The code that actually uses this list looks similar to the fix for kelloggm/checker-framework#14.  Should this special handling be removed once that issue is fixed?  If so, please add a comment to that issue.
`IndexFirstArgListMethods` should be `indexFirstArgListMethods`.  Also, please add a comment.